Sources/MongoSwift/CursorCommon.swift

```diff
@@ -1,7 +1,10 @@
+import CLibMongoC
```

Comment:
The goal of this refactor was to separate concerns and consolidate common code. To that end, the following internal types were introduced/tweaked:
- `protocol CursorProtocol`: this is the existing protocol (formerly named just `Cursor`) that defines shared behavior between our public cursor types. This now defines solely the common _public API_ between cursor like things.
- ` protocol MongocCursorWrapper`: new protocol defining behavior for direct swift wrappers of mongoc types. The two types that conform to this are `MongocCursor` and `MongocChangeStream`. This protocol allows us to use functions like `mongoc_change_stream_next` abstractly.
- `class Cursor<CursorKind: MongocCursor>`: This type contains all the logic that is currently duplicated between the existing `ChangeStream` and `MongoCursor` implementations. It's generic over the type of the mongoc wrapper.
    - Each of `MongoCursor` and `ChangeStream` now wrap one of these internal `Cursor<T>`s, and implement their public API methods by calling different methods on it. We can think of them as consumers of our internal cursor API.